File: examples/nodejs/subscribe_candlesticks.js
const { Config, QuoteContext, Period, TradeSessions } = require("longport");

let globalCtx;

async function main() {
  let config = Config.fromEnv();
  globalCtx = await QuoteContext.new(config);
  globalCtx.setOnCandlestick((_, event) => console.log(event.toString()));
  globalCtx.subscribeCandlesticks(
    "AAPL.US",
    Period.Min_1,
    TradeSessions.Normal
  );
}

Promise.all([main()]).catch((err) => console.error(err));

File: examples/python/subscribe_candlesticks.py
from time import sleep
from longport.openapi import QuoteContext, Config, Period, PushCandlestick


def on_candlestick(symbol: str, event: PushCandlestick):
    print(symbol, event)


config = Config.from_env()
ctx = QuoteContext(config)
ctx.set_on_candlestick(on_candlestick)

ctx.subscribe_candlesticks("AAPL.US", Period.Min_1)
sleep(30)
